gemrb/core/PathFinder.cpp

```diff
@@ -259,15 +270,35 @@ Path Map::FindPath(const Point& s, const Point& d, unsigned int size, unsigned i
 	if (!mapSize.PointInside(smptSource)) return {};
 
 	// Initialize data structures
-	FibonacciHeap<PQNode> open;
-	std::vector<bool> isClosed(mapSize.Area(), false);
-	std::vector<NavmapPoint> parents(mapSize.Area(), Point(0, 0));
-	std::vector<unsigned short> distFromStart(mapSize.Area(), std::numeric_limits<unsigned short>::max());
+	const size_t mapCellsCount = mapSize.Area();
+
+	FibonacciHeap<PQNode> open; // todo: remove this data structure or rewrite its implementation - too much allocations there!
+	// make most data storage for this algorithm static, to avoid memory allocations;
+	// each run we just clear the storage, which is keeping the underlying allocated memory at hand
+	static std::vector<bool> isClosed(mapSize.Area(), false);
+	static std::vector<NavmapPoint> parents(mapSize.Area(), Point(0, 0));
+	static std::vector<unsigned short> distFromStart(mapSize.Area(), std::numeric_limits<unsigned short>::max());
+
+	// resize if needed (in case of a map change; probably can be done once, when new map is loaded)
+	if (isClosed.size() != mapCellsCount) {
+		isClosed.resize(mapCellsCount);
```

Comment:
Isn't this redundant? You always resize it below.

Reply:
Eagle eye! Thanks.